### PR TITLE
fix oracle strips learning when options are unknown

### LINKED ISSUE #762: oracle strips learner crashes when options are unknown
to reproduce:
```
python src/main.py --env cover --approach nsrt_learning --strips_learner oracle --option_learner oracle --seed 0
```
error:
```
Traceback (most recent call last):
  File "/Users/tom/phd/predicators/src/main.py", line 356, in <module>
    main()
  File "/Users/tom/phd/predicators/src/main.py", line 104, in main
    _run_pipeline(env, approach, stripped_train_tasks, offline_dataset)
  File "/Users/tom/phd/predicators/src/main.py", line 126, in _run_pipeline
    approach.learn_from_offline_dataset(offline_dataset)
  File "/Users/tom/phd/predicators/src/approaches/nsrt_learning_approach.py", line 48, in learn_from_offline_dataset
    self._learn_nsrts(dataset.trajectories, online_learning_cycle=None)
  File "/Users/tom/phd/predicators/src/approaches/nsrt_learning_approach.py", line 52, in _learn_nsrts
    self._nsrts = learn_nsrts_from_data(
  File "/Users/tom/phd/predicators/src/nsrt_learning/nsrt_learning_main.py", line 61, in learn_nsrts_from_data
    pnads = learn_strips_operators(
  File "/Users/tom/phd/predicators/src/nsrt_learning/strips_learning/__init__.py", line 46, in learn_strips_operators
    return learner.learn()
  File "/Users/tom/phd/predicators/src/nsrt_learning/strips_learning/base_strips_learner.py", line 42, in learn
    learned_pnads = self._learn()
  File "/Users/tom/phd/predicators/src/nsrt_learning/strips_learning/oracle_learner.py", line 27, in _learn
    self._recompute_datastores_from_segments(pnads)
  File "/Users/tom/phd/predicators/src/nsrt_learning/strips_learning/base_strips_learner.py", line 195, in _recompute_datastores_from_segments
    segment_option = segment.get_option()
  File "/Users/tom/phd/predicators/src/structs.py", line 1107, in get_option
    assert self.has_option()
AssertionError
```

### PR DESCRIPTION
**Pull request by tomsilver**
closes #762

### CODE REVIEW

File: src/nsrt_learning/strips_learning/base_strips_learner.py [hunk 1]
```diff
@@ -192,7 +192,9 @@ def _recompute_datastores_from_segments(
         for seg_traj in self._segmented_trajs:
             objects = set(seg_traj[0].states[0])
             for segment in seg_traj:
-                segment_option = segment.get_option()
+                segment_option = DummyOption
```
Comment (ronuchit) [inline src/nsrt_learning/strips_learning/base_strips_learner.py:195]: wanna do this as an if else like your change to oracle_learner.py?

File: tests/nsrt_learning/strips_learning/test_oracle_learner.py [hunk 2]
```diff
@@ -34,3 +34,36 @@ def test_oracle_strips_learner():
     Delete Effects: [Clear(?block:block), GripperOpen(?robot:robot), On(?block:block, ?otherblock:block)]
     Side Predicates: []
     Option Spec: Pick(?robot:robot, ?block:block)]"""  # pylint: disable=line-too-long
+    # Test with unknown options. Expected behavior is that the operators should
```
Comment (ronuchit) [inline tests/nsrt_learning/strips_learning/test_oracle_learner.py:37]: did you verify that this test was crashing before the change?

**Review by ronuchit — APPROVED** [review]

File: src/nsrt_learning/strips_learning/base_strips_learner.py (hunk 1 above)
Comment (tomsilver) [reply src/nsrt_learning/strips_learning/base_strips_learner.py:195]: I had that originally, but then getting test coverage was being annoying and led to rabbit holes.. I'll give it another shot though

File: tests/nsrt_learning/strips_learning/test_oracle_learner.py (hunk 2 above)
Comment (tomsilver) [reply tests/nsrt_learning/strips_learning/test_oracle_learner.py:37]: yup

**Review by ronuchit — APPROVED** [review]

**Pull request merged**